### PR TITLE
Update Hess product function

### PR DESCRIPTION
**Pull request by liside**

### CODE REVIEW

File: morpheus/normalized_matrix.py [hunk 1]
```diff
@@ -575,6 +575,164 @@ def _cross(self, matrix_a, matrix_b=None):
...
+                            res[ni1:ni2, dj1:dj2] = r[i].T.dot(m.T.dot(r[j]))
+                            res[dj1:dj2, ni1:ni2] = res[ni1:ni2, dj1:dj2].T
+                        else:
+                            res[ni1:ni2, dj1:dj2] = (w.reshape(-1, 1) * np.array(r[i][k[i]])).T.dot(r[j][k[j]])
+                            res[dj1:dj2, ni1:ni2] = res[ni1:ni2, dj1:dj2].T
+
+        return res
+    
+    """
+    """
```
Comment (liside) [inline morpheus/normalized_matrix.py:735]: unnecessary comments

File: morpheus/normalized_matrix.py [hunk 2]
```diff
@@ -575,6 +575,164 @@ def _cross(self, matrix_a, matrix_b=None):
 
     def dot(self, other):
         return self.__mul__(other)
+    
+    """
+    Change
+    """
+
```
Comment (liside) [inline morpheus/normalized_matrix.py:582]: maybe a few lines explaining what this function is used for?

File: morpheus/normalized_matrix.py [hunk 3]
```diff
@@ -575,6 +575,164 @@ def _cross(self, matrix_a, matrix_b=None):
...
+            count = ds
+            cross_r = []
+            for t in r:
+                if all(map(sp.issparse, r)):
+                    cross_r.append(self._t_cross_w(t, w[count:count+t.shape[1]]).toarray())
+                else:
+                    cross_r.append(self._t_cross_w(t, w[count:count+t.shape[1]]))
+
+
```
Comment (liside) [inline morpheus/normalized_matrix.py:629]: empty line

**Review by liside — COMMENTED** [review]
Overall LGTM. Please remove necessary comments and empty lines (there are two or three empty lines even in one function). Also, can you add a simple unit test case to make sure it works properly?

**Comment by liside** [comment]
thanks for the update. can you add a unit test case in `test_normalized_matrix.py`? using your example in the function comment would be sufficient

**Comment by shy218** [comment]
I have updated the test case in test_normalized_matrix.py for normal and
transpose cases. Thank you for taking the time to check.

**Pull request merged**